app/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp

```diff
@@ -70,15 +70,16 @@
                 <form:password cssClass="form-control" cssErrorClass="error" id="password" placeholder= "${passwordPlaceholder}" size="25" tabindex="2" path="password"  accesskey="${passwordAccessKey}" htmlEscape="true" autocomplete="off" />
                 <spring:message code="screen.password.forgotText" var="passwordForgotText" text="" javaScriptEscape="true"/>
                 <c:if test="${not empty passwordForgotText}">
-                <a id="forgotPassword" class="link-underlined" onclick="toggleForgotPasswordInfo()"><spring:message code="screen.password.forgot"/></a>
+                    <spring:message code="screen.password.forgot" var="passwordForgot" text="" javaScriptEscape="true"/>
+                    <button id="forgotPassword" class="link-underlined btn btn-primary" aria-label="${passwordForgot}" onclick="toggleForgotPasswordInfo()">${passwordForgot}</button>
                 </c:if>
             </div>
-            
+
             <input type="hidden" name="lt" value="${loginTicket}" />
             <input type="hidden" name="execution" value="${flowExecutionKey}" />
             <input type="hidden" name="_eventId" value="submit" />
 
-            <input class="btn btn-primary pull-right" name="submit" accesskey="l" value="Login" tabindex="4" type="submit" />
+            <input class="btn btn-primary pull-right" role="button" name="submit" accesskey="l" value="Login" tabindex="4" type="submit" />
```

Comment:
The semantically better element is a button. An input should only be used to receive input from the user.